src/Microsoft.DotNet.Build.Tasks/GenerateResourcesCode.cs

```diff
@@ -92,91 +88,116 @@ private void WriteClassHeader()
                 _targetStream.WriteLine("    Friend Partial Class SR");
                 _targetStream.WriteLine("    ");
 
-                _targetStream.WriteLine("        Private Const s_resourcesName As String = \"{0}\"", _resourcesName + ".SR");
+                _targetStream.WriteLine($"        Private Const s_resourcesName As String = \"{_resourcesName}.SR\"");
                 _targetStream.WriteLine("");
-                if (!DebugOnly)
-                    _targetStream.WriteLine("#If Not DEBUGRESOURCES Then");
             }
         }
 
-        private void RunOnResFile()
+        private void WriteResources()
         {
-            foreach(KeyValuePair<string, string> pair in GetResources(ResxFilePath))
+            var resources = GetResources(ResxFilePath);
+
+            if (AsConstants)
             {
-                StoreValues((string)pair.Key, (string)pair.Value);
+                foreach (var resourcePair in resources)
+                {
+                    WriteResourceConstant((string)resourcePair.Key);
+                }
+            }
+            else
+            {
+                _targetStream.WriteLine(_targetLanguage == TargetLanguage.CSharp ?
+                    "#if !DEBUGRESOURCES" :
```

Comment:
>DEBUGRESOURCES [](start = 26, length = 14)

I am not sure if we still need this either. This would be useful for .Net Native only and I don't think it is used any more as you have removed the Debug property anyway.

Reply:
This is definitely still used in corefx. It keeps the literal resource strings out of the il in ret builds.

Reply:
What I mean we always produce the code without literal resource strings. so we'll not produce the parts under DEBUGRESOURCES.

Reply:
it is ok to keep for now though so you don't have to block on that.

Reply:
I confirmed with Morgan than we don't need to generate the literal strings anymore. this was used before with .Net Native but it is not needed anymore.

Reply:
if you remove it, you will need to remove the line 

https://github.com/dotnet/buildtools/blob/63db91f2fe1a7aec78ff190e7307dbdc0b62757d/src/Microsoft.DotNet.Build.Tasks/PackageFiles/resources.targets#L16

too. Thanks

Reply:
I'm not willing to bet on that just yet, so I'll hold off on that cleanup.